Data/Scripts/Items/Houses/Construction/Addons/ArcheryButteAddon.cs

```diff
@@ -178,13 +178,14 @@ public void Fire( Mobile from )
 			}
 
 			Container pack = from.Backpack;
+			Container quiver = from.FindItemOnLayer(Layer.Cloak) as Container;
 			Type ammoType = bow.AmmoType;
 
 			bool isArrow = ( ammoType == typeof( Arrow ) );
 			bool isBolt = ( ammoType == typeof( Bolt ) );
 			bool isKnown = ( isArrow || isBolt );
 
-			if ( from is PlayerMobile && ( pack == null || !pack.ConsumeTotal( ammoType, 1 ) ) )
+			if ( from is PlayerMobile && (( pack == null || !pack.ConsumeTotal( ammoType, 1 ) && (quiver == null || !quiver.ConsumeTotal( ammoType, 1) ) ) ) )
```

Comment:
I'll admit, this was hard to read, but I wrapped my head around it eventually.

Looks good in terms of functionality but might be nice to spread it out into some bool vars above for future devs to read. 😅

Reply:
feel free to suggest changes in here!

Reply:
Oh right! We can do that now!

```suggestion

			// Display error message if player cannot consume ammo from pack or quiver
			bool cannotConsumeFromPack = pack == null || !pack.ConsumeTotal( ammoType, 1 );
			bool cannotConsumeFromQuiver = quiver == null || !quiver.ConsumeTotal( ammoType, 1);

			if ( from is PlayerMobile && cannotConsumeFromPack && cannotConsumeFromQuiver )
```

Untested as I did this right in GitHub but should work! 🤞